opentelemetry-sdk/src/logs/record.rs

```diff
@@ -34,7 +44,7 @@ pub struct LogRecord {
     pub body: Option<AnyValue>,
 
     /// Additional attributes associated with this record
-    pub attributes: Option<Vec<(Key, AnyValue)>>,
+    pub attributes: AttributesGrowableArray,
```

Comment:
Does this force `GrowableArray` to be public? Can we avoid that?

Reply:
`GrowableArray` is public for now. We can make it private, and only keep it's iterators public, which can be used by the exporter.

Reply:
Yes please. We certainly don't want to leak any implementation information public.